src/sagas/index.js

```diff
@@ -15,23 +15,35 @@ import { saga as permissionsSagas } from './permissions';
 import { saga as walletConnectSagas } from './walletConnect';
 import { saga as networkSettingsSagas } from './networkSettings';
 import { saga as nanoContractSagas } from './nanoContract';
+import { logger } from '../logger';
+import { onExceptionCaptured } from '../actions';
 
-const sagas = [
-  walletSagas,
-  tokensSagas,
-  pushNotificationSaga,
-  networkSettingsSagas,
-  errorHandlerSagas,
-  featureToggleSagas,
-  permissionsSagas,
-  walletConnectSagas,
-  nanoContractSagas,
-];
+function* rootSaga() {
```

Comment:
This could be a PR on its own, the idea here is to re-create the saga if some unhandled error bubble up to the rootSaga

The current behavior is that if any saga crashes, the root saga crashes and nothing works anymore

Reply:
I'm doing this in this PR because things might fail unexpectedly because of wallet-connect, so this prevents it from propagating to other sagas